### DISCUSSION TITLE
Embedding external language

### DISCUSSION THREAD

**Discussion by scripthunter7** [github-linguist/linguist#6020] [Q&A] [answered]
Hello!

I would like to ask is Linguist supports embedding an external language in my own syntax highlighter? I'd like to embed Javascript in our tmlanguage according to the example below:

`package.json`:
```json
"grammars": [
    {
        "language": "adblock",
        "scopeName": "text.adblock",
        "path": "./syntaxes/adblock.tmLanguage.json",
        "embeddedLanguages": {
            "source.js": "javascript"
        }
    }
]
```

`adblock.tmLanguage.json`:
```json
{
    ...
    "contentName": "source.js",
    "patterns": [
        {
            "include": "source.js"
        }
    ]
}
```

This works in VSCode, but is there any extra TODO required to make it work in Linguist? Or is it possible at all? Although, HTML works in Linguist.

**Comment by lildude** [comment, marked as answer]
There's nothing extra to do provided Linguist includes the scope you're referencing, which in this case it does. It should work as it does in VSCode.
